manifests/extension.pp

```diff
@@ -57,161 +53,46 @@
-  $zend              = false,
-  $settings          = {},
-  $settings_prefix   = false,
-  $sapi              = 'ALL',
-  $responsefile      = undef,
+  String           $ensure            = 'installed',
+  Optional[Php::Provider] $provider   = undef,
+  Optional[String] $source            = undef,
+  Optional[String] $so_name           = downcase($name),
+  Optional[String] $php_api_version   = undef,
+  String           $package_prefix    = $::php::package_prefix,
+  Boolean          $zend              = false,
+  Hash             $settings          = {},
+  Php::Sapi        $sapi              = 'ALL',
+  Variant[Boolean, String]       $settings_prefix   = false,
+  Optional[Stdlib::AbsolutePath] $responsefile      = undef,
+  Variant[String, Array[String]] $header_packages   = [],
+  Variant[String, Array[String]] $compiler_packages = $::php::params::compiler_packages,
 ) {
 
   if ! defined(Class['php']) {
     warning('php::extension is private')
   }
 
-  validate_string($ensure)
-  validate_string($package_prefix)
-  validate_string($so_name)
-  validate_string($php_api_version)
-  validate_string($sapi)
-  validate_array($header_packages)
-  validate_bool($zend)
-
-  if $source and $pecl_source {
-    fail('Only one of $source and $pecl_source can be specified.')
-  }
-
-  if $source {
-    $real_source = $source
-  }
-  else {
-    $real_source = $pecl_source
-  }
-
-  if $provider != 'none' {
-    case $provider {
-      'pecl': { $real_package = $title }
-      'pear': { $real_package = $title }
-      default: { $real_package = "${package_prefix}${title}" }
-    }
-
-    unless empty($header_packages) {
-      ensure_resource('package', $header_packages)
-      Package[$header_packages] -> Package[$real_package]
-    }
-
-    if $provider == 'pecl' or $provider == 'pear' {
-      package { $real_package:
-        ensure       => $ensure,
-        provider     => $provider,
-        source       => $real_source,
-        responsefile => $responsefile,
-        require      => [
-          Class['::php::pear'],
-          Class['::php::dev'],
-        ],
-      }
-
-      unless empty($compiler_packages) {
-        ensure_resource('package', $compiler_packages)
-        Package[$compiler_packages] -> Package[$real_package]
-      }
-    }
-    else {
-      if $responsefile != undef {
-        warning("responsefile param is not supported by php::extension provider ${provider}")
-      }
-
-      package { $real_package:
-        ensure   => $ensure,
-        provider => $provider,
-        source   => $real_source,
-      }
-    }
-
-    $package_depends = "Package[${real_package}]"
-  } else {
-    $package_depends = undef
+  php::extension::install { $title:
+    ensure            => $ensure,
+    provider          => $provider,
+    source            => $source,
+    responsefile      => $responsefile,
+    header_packages   => $header_packages,
+    compiler_packages => $compiler_packages,
   }
```

Comment:
failing tests are due to package_prefix not being passed to extension::install

tested locally, rake spec is fine on centos6/7 and rhel6/7